test/org/sagebionetworks/bridge/services/HealthDataServiceTest.java

```diff
@@ -123,4 +140,43 @@ public void getRecordsForUploadDateSuccess() {
         assertEquals("bar healthcode", recordList.get(1).getHealthCode());
         assertEquals("baz healthcode", recordList.get(2).getHealthCode());
     }
+
+    @Test
+    public void updateRecordSuccess() throws Exception {
+
+        // first create a mock record
+        // record
+        HealthDataRecord record = DAO.getRecordBuilder().withHealthCode(TEST_HEALTH_CODE)
+                .withSchemaId(TEST_SCHEMA_ID).withSchemaRevision(3).withStudyId(TEST_STUDY_ID).build();
+
+        // mock dao
+        HealthDataDao mockDao = mock(HealthDataDao.class);
+        when(mockDao.createOrUpdateRecord(record)).thenReturn(TEST_RECORD_ID);
+
+        HealthDataService svc = new HealthDataService();
+        svc.setHealthDataDao(mockDao);
+
+        // execute and validate
+        String retVal = svc.createOrUpdateRecord(record);
+        assertEquals(TEST_RECORD_ID, retVal);
+
+        // then create a mock json request
+        RecordExportStatusRequest recordExportStatusRequest = createMockRecordExportStatusRequest();
+        when(mockDao.getRecordById(TEST_RECORD_ID)).thenReturn(record);
+
+        // finally call service method and assert
+        svc.updateRecordsWithExporterStatus(recordExportStatusRequest);
+        HealthDataRecord recordAfter = svc.getRecordById(TEST_RECORD_ID);
+        assertNotNull(recordAfter.getSynapseExporterStatus());
+        assertEquals(HealthDataRecord.ExporterStatus.SUCCEEDED, recordAfter.getSynapseExporterStatus());
```

Comment:
I don't think this is verifying correctly. You just mocked this return value you are verifying, which only tells you that your mock worked. You want to verify() that the DAO was called with the correct object, that is, that this service performed the right call through to the DAO.

Also, please write tests to validate that all the error conditions you are checking for in the service are correctly being validated, this is the key work it is doing: validating all the inputs and communicating errors back to the user that allow them to fix the problem.